iuf

```diff
@@ -313,8 +312,10 @@ def get_answer():
     """Read input and handle related exceptions."""
     answer = ""
     try:
+        RLOCK.acquire()
         answer = input()
-    except RuntimeError:
+        RLOCK.release()
```

Comment:
If there's an exception this will never release.   Since we don't actually care about the exception from the looks of it maybe add a 

Finally:
   RLOCK.release()

so it always happens

Reply:
Nice catch.  Thanks!